### ISSUE TITLE
Logout does not remove current user from UserStore

### ISSUE THREAD

**Issue by joluet** [realm/realm-java#3866]
Steps to reproduce:
0. Make sure a user is logged in
1. Call `SyncUser.currentUser().logout()` 
2. Kill the app manually
3. Restart the app
4. `SyncUser.currentUser()` still returns the user that should have been logged out

This looks like it could be the reason:
`SyncUser.logout()` tries to remove the current user from the `UserStore` like this:
```
if (SyncUser.this.equals(SyncUser.currentUser())) {
   SyncManager.getUserStore().remove(UserStore.CURRENT_USER_KEY);
}
```
However, at that point `SyncUser.currentUser()` returns `null` because the user's token has already been removed in the line before and thus `isValid()` will return `false`. Therefore,  `remove(UserStore.CURRENT_USER_KEY)` is not executed.

**Comment by joluet** [comment]
This worked for me as a workaround:


```
package io.realm;

public class LogoutHelper {

    public static void deleteCurrentUser() {
        SyncManager.getUserStore().remove(UserStore.CURRENT_USER_KEY);
    }
}
```

Then I do this to log the current user out:
```
SyncUser.currentUser().logout();
LogoutHelper.deleteCurrentUser();
```

**Comment by kneth** [comment]
I wonder if it is related to https://github.com/realm/realm-object-store/issues/260.

And I believe it will be fixed by #3741.

**Comment by joluet** [comment]
I don't think that issue is related to this one. As I explained above I think that the code in `SyncUser` does not work as intended. There is an if statement which prevents the current user from being removed from the `UserStore` because its condition is always false.

**Comment by kneth** [comment]
@nhachicha Do you have any thoughts on this?

**Comment by nhachicha** [comment]
Hi @joluet 
The new (_default_) implementation on [master](https://github.com/realm/realm-java/blob/master/realm/realm-library/src/objectServer/java/io/realm/SyncUser.java#L268) of the UserStore is deleting the current user directly, this should fix your issue I believe.

**Comment by beeender** [comment]
This should be fixed by #3838 and will be released in `v2.3.0`. The snapshot release should contain this fix now.
